crates/algokit_utils/tests/transactions/composer/key_registration.rs

```diff
@@ -0,0 +1,497 @@
+        .expect("Failed to add online key registration");
+
+    // Submit the transaction - should succeed with proper keys and voting rounds
+    let result = composer
+        .send()
+        .await
+        .expect("Failed to send online key registration");
+
+    // Assert transaction was confirmed
+    assert!(
+        result.confirmed_round.is_some(),
+        "Transaction should be confirmed"
+    );
+    assert!(
+        result.confirmed_round.unwrap() > 0,
+        "Confirmed round should be greater than 0"
+    );
+
+    let transaction = result.txn.transaction;
+
+    match transaction {
+        algokit_transact::Transaction::KeyRegistration(key_reg_fields) => {
+            assert_eq!(
+                key_reg_fields.vote_key,
+                Some(vote_key),
+                "Vote key should match"
+            );
+            assert_eq!(
+                key_reg_fields.selection_key,
+                Some(selection_key),
+                "Selection key should match"
+            );
+            assert_eq!(
+                key_reg_fields.vote_first,
+                Some(vote_first),
+                "Vote first should match"
+            );
+            assert_eq!(
+                key_reg_fields.vote_last,
+                Some(vote_last),
+                "Vote last should match"
+            );
+            assert_eq!(
+                key_reg_fields.vote_key_dilution,
+                Some(100),
+                "Vote key dilution should match"
+            );
+            assert_eq!(
+                key_reg_fields.state_proof_key,
+                Some(state_proof_key),
+                "State proof key should match"
+            );
+            assert!(
+                key_reg_fields.non_participation.is_none(),
+                "Non participation should be None for online registration"
+            );
+        }
+        _ => panic!("Transaction should be a key registration transaction"),
+    }
+
+    // Verify account participation status
+    let account_info = context
+        .algod
+        .account_information(None, &sender_addr.to_string(), None)
+        .await
+        .expect("Failed to get account information");
+
+    // For online registration, participation should contain the keys
+    if let Some(participation) = account_info.participation {
+        assert!(
+            !participation.vote_participation_key.is_empty(),
+            "Account should have participation keys after going online"
+        );
+
+        // Verify the participation keys match what we submitted
+        assert_eq!(
+            participation.vote_participation_key,
+            vote_key.to_vec(),
+            "Vote participation key should match submitted key"
+        );
+    } else {
+        panic!("Account should have participation information after online key registration");
+    }
+}
+
+#[tokio::test]
+async fn test_permanent_offline_key_registration_transaction() {
```

Comment:
This is technically the same as non participation key registration. Probably makes sense to fold any useful logic into that test.